SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/ConnectionConfig.h

```diff
@@ -176,27 +174,31 @@ class FReceptionistConfig : public FConnectionConfig
 
 	bool TryLoadCommandLineArgs()
 	{
-		bool bSuccess = true;
 		const TCHAR* CommandLine = FCommandLine::Get();
 
+		// Get command line options first since the URL handling will modify the CommandLine string
+		FParse::Value(CommandLine, TEXT("receptionistPort"), ReceptionistPort);
+
 		// Parse the command line for receptionistHost, if it exists then use this as the host IP.
-		if (!FParse::Value(CommandLine, TEXT("receptionistHost"), ReceptionistHost))
+		FString Host;
+		if (!FParse::Value(CommandLine, TEXT("receptionistHost"), Host))
 		{
 			// If a receptionistHost is not specified then parse for an IP address as the first argument and use this instead.
 			// This is how native Unreal handles connecting to other IPs, a map name can also be specified, in this case we use the default IP.
 			FString URLAddress;
-			FParse::Token(CommandLine, URLAddress, 0);
-			FRegexPattern Ipv4RegexPattern(TEXT("^(?:[0-9]{1,3}\\.){3}[0-9]{1,3}$"));
-			FRegexMatcher IpV4RegexMatcher(Ipv4RegexPattern, *URLAddress);
-			bSuccess = IpV4RegexMatcher.FindNext();
-			if (bSuccess)
+			FParse::Token(CommandLine, URLAddress, false);
+			const FURL URL(nullptr, *URLAddress, TRAVEL_Absolute);
+			if (URL.Valid && !URL.Host.IsEmpty())
 			{
-				SetReceptionistHost(URLAddress);
+				SetReceptionistHost(URL);
 			}
 		}
+		else
+		{
+			SetReceptionistHost(Host);
```

Comment:
Will this Host just be empty? In which case do we want to set ReceptionistHost to an empty string or do we want to use the default host? Perhaps we could guard against this by checking inside SetReceptionistHost that the input host is not empty?

Reply:
Good catch

Reply:
Do we need to call this here? Is this not the case where Host is empty?

Reply:
Just thought I would bump this comment

Reply:
The if condition is inverted. If you get to the else, then you have successfully read from the `receptionistHost` parameter

Reply:
Oh lol! I am just a chimp. Sweet! Looks good!